### ISSUE TITLE
Feature Request: No Cooling While Ironing

### ISSUE THREAD

**Issue by Fuzzyl0g1x** [SoftFever/OrcaSlicer#6979]
### Is there an existing issue for this feature request?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

When ironing at low volumetric flow rates, nozzle clogs / extruder jams are common and quality could be better.

### Which printers will be beneficial to this feature?

All

### Describe the solution you'd like

To achieve the smoothest ironing possible, Orca should have a checkbox named, "disable cooling fans while ironing" to allow the filament to stay warm and flow into the crevices of the top surface.  This will provide a benefit to all users, regardless of the nozzle size selected.

Disabling cooling while ironing is also important because when you're ironing, your volumetric flow rate is extremely low and excessive cooling air can cause a clog by flashing the filament solid after it leaves the tip.

This is a common problem when ironing with a 0.2 nozzle: 0.06 layer height x 0.22 line width x 30 mm/s x 30% ironing flow = 0.1188 mm3 /s. Experimentally, I have found that printing below 0.20-0.25 mm3 /s guarantees a clog followed by extruder jam on the Bambu P1S if you don't catch it in time.  Manually reducing airflow can help prevent this, but it is not always possible to be available when a print begins ironing.

**Comment by aidan-gibson** [comment]
In the meantime, this is how I did it automatically:

<img width="751" alt="image" src="https://github.com/user-attachments/assets/ea019a5a-7f7b-4d2b-9f01-13818ba17f9d">

**Comment by hotellonely** [comment]
> In the meantime, this is how I did it automatically:
> 
> <img alt="image" width="751" src="https://private-user-images.githubusercontent.com/20917687/375958031-ea019a5a-7f7b-4d2b-9f01-13818ba17f9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAyNTkzMTIsIm5iZiI6MTczMDI1OTAxMiwicGF0aCI6Ii8yMDkxNzY4Ny8zNzU5NTgwMzEtZWEwMTlhNWEtN2Y3Yi00ZDJiLTlmMDEtMTM4MThiYTE3ZjlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDMwVDAzMzAxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhMGU3MzIwNTM3MWFjYzFlOTc3ODI4MjYwMTg2ZmE3MTk2MDNhOTUyZGM3MzgzODllMmY0MWI3NTc2MTQ2ODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.tk_G9de25RBLzQJSAlHpXQp4EKLb1VexjsBgiKCysUk">

This is awesome, thanks!

**Comment by fuorissimo** [comment]
> In the meantime, this is how I did it automatically:
> <img alt="image" width="751" src="https://private-user-images.githubusercontent.com/20917687/375958031-ea019a5a-7f7b-4d2b-9f01-13818ba17f9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NzgzNDAsIm5iZiI6MTczMDY3ODA0MCwicGF0aCI6Ii8yMDkxNzY4Ny8zNzU5NTgwMzEtZWEwMTlhNWEtN2Y3Yi00ZDJiLTlmMDEtMTM4MThiYTE3ZjlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDIzNTQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjYTIxNDFmMWY3YjRlNGVhOTZiZWE2MjA5OWI4ODc5ZTMzNTJkOTk2MzE1NGRiYmNmZjkzM2Q3NjdjZTc3MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jz_jps8QJIU-_S--yi6oyAFQdPT94bFmBmgLoVXqetM">

Your code leaves the fans off even after ironing. I use this but it's not good anyway because the fans remain constant and don't stay off for example in the first layers or completely when printing PETG. A specific Orca function is required.

{if extrusion_role=~/Ironing/}
  M107 ; disable fan
  ; M107 ; disable additional fan
{else}
  M106 S[fan_min_speed] ; reactivate fans
  M106 P2 S[fan_min_speed] ; reactivate additional fan
{endif}

**Comment by aidan-gibson** [comment]
Yep you are totally correct, I ended up doing the same, I should've updated here. I don't love this bodge. Wish there was a way to set fans to previous speed instead of picking between `fan_min_speed` and `fan_max_speed`; hopefully we'll see this feature natively integrated at some point. Documentation on using variables etc in the Machine G-code section of Orca would also be so helpful, I've just been code searching this repo and only half knowing what I'm doing.

**Comment by fuorissimo** [comment]
> Yep you are totally correct, I ended up doing the same, I should've updated here. I don't love this bodge. Wish there was a way to set fans to previous speed instead of picking between `fan_min_speed` and `fan_max_speed`; hopefully we'll see this feature natively integrated at some point. Documentation on using variables etc in the Machine G-code section of Orca would also be so helpful, I've just been code searching this repo and only half knowing what I'm doing.

I have to say thank you because your suggestion solved the problem, I no longer have those artifacts. On the other hand, however, now you can see the filling underneath, which didn't happen before despite using 4 layers. Probably the best solution is not to turn off the fan completely during ironing but to slow it down. Now I do a test at 25% and 50%. Meanwhile, here is the result with the fan at 100% (full of lumps) and turned off.
![WhatsApp Image 2024-11-04 at 03 55 47(1)](https://github.com/user-attachments/assets/f134860b-e578-48dd-87f8-8048cb956473)

**Comment by fuorissimo** [comment]
*** IRONING UPDATE ***
I did a lot of tests with an open KOBRA 2 MAX printer and PLA+ material (Jayo/Sunlu). With closed printers it is possible that better cooling is needed.

It seems that by turning off the fan completely you get a shinier and smoother surface but you can glimpse the underlying filling even if you increase the thickness. The tests in the photo refer to 10% gyroid infill and 4 levels (0.8mm) higher but even increasing to 15% infill and 5 layers higher the result does not change.

The best result is obtained with a fan between 20% and 25%. It is slightly less smooth to the touch but the underlying filling almost completely disappears. In my opinion the 25% fan is the best solution for an open printer.
![WhatsApp Image 2024-11-05 at 07 19 52](https://github.com/user-attachments/assets/355152e7-84b0-41f2-a1ad-45b01a3b4988)
![WhatsApp Image 2024-11-05 at 07 19 51(2)](https://github.com/user-attachments/assets/e71b7497-02f6-485e-b836-4ba1f08867e8)
![WhatsApp Image 2024-11-05 at 07 19 50](https://github.com/user-attachments/assets/781b6c5a-ca54-411c-bd19-85c80734dba8)

**Comment by fuorissimo** [comment]
*** IRONING UPDATE ***
I did a lot of tests with an open KOBRA 2 MAX printer and PLA+ material (Jayo/Sunlu). With closed printers it is possible that better cooling is needed.

It seems that by turning off the fan completely you get a shinier and smoother surface but you can glimpse the underlying filling even if you increase the thickness. The tests in the photo refer to 10% gyroid infill and 4 top layers (0.8mm) but even increasing to 15% infill and 5 top layers the result does not change.

The best result is obtained with a fan between 20% and 25%. It is slightly less smooth to the touch but the underlying filling almost completely disappears. In my opinion the 25% fan is the best solution for an open printer.

At the moment the only way is to enter this code in Orca Slicer (and its variants) in the Printer Settings > Machine Gcode section:

{if extrusion_role=~/Ironing/}
  M106 S64 ; slow down fan to 25%
  M106 P2 S64 ; slow down additional fans to 25%
{else}
  M106 S255 ; reactivate fan to 100%
  M106 P2 S255 ; reactivate additional fan to 100%
{endif}

However this code creates some small problems. If you set the fans (in filament settings) to stop in the first X layers for some reason at the first layer the fan is at 100%. It is necessary to stop it manually from the printer monitor as soon as the printing of the first layer filling begins. Furthermore, once the ironing phase is completed the fan will always be at 100% so this must be kept in mind if you are printing PETG for example. 

We hope that Orca Slicer will allow to choose the fan speed during the ironing phase in the future.


![WhatsApp Image 2024-11-05 at 07 19 50](https://github.com/user-attachments/assets/ac205454-1cd3-4cc5-af6a-702e1a8e001e)
![WhatsApp Image 2024-11-05 at 07 19 52](https://github.com/user-attachments/assets/a0538824-4010-426a-a20f-f58e06738e7f)
![WhatsApp Image 2024-11-05 at 07 19 51(2)](https://github.com/user-attachments/assets/e203cc71-06ec-4aa8-9d97-409091d0e293)
![Screenshot (742)](https://github.com/user-attachments/assets/5aaca52c-9391-436f-b741-cb4e2bd6bef3)

**Comment by Fuzzyl0g1x** [comment]
[fuorissimo](https://github.com/fuorissimo), if I'm understanding correctly, < 10% cooling while ironing with PLA (which has a low melting temperature) and very sparse infill (you are using 10%) can cause the infill to become visible through the top layers.  This is because the hotend is slightly remelting the top layers, causing them to sag into the infill beneath.  Other materials, which are more heat resistant, or different print settings (e.g. higher density infill) might be able to tolerate less cooling while ironing.

To summarize, I think you did a great job proving your point, which is that a numerical setting is needed, not simply a checkbox that disables cooling.  Rather than "no cooling while ironing" there should be a "reduced cooling while ironing" setting.

**Comment by Wewe12** [comment]
> > In the meantime, this is how I did it automatically:
> > <img alt="image" width="751" src="https://private-user-images.githubusercontent.com/20917687/375958031-ea019a5a-7f7b-4d2b-9f01-13818ba17f9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NzgzNDAsIm5iZiI6MTczMDY3ODA0MCwicGF0aCI6Ii8yMDkxNzY4Ny8zNzU5NTgwMzEtZWEwMTlhNWEtN2Y3Yi00ZDJiLTlmMDEtMTM4MThiYTE3ZjlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDIzNTQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjYTIxNDFmMWY3YjRlNGVhOTZiZWE2MjA5OWI4ODc5ZTMzNTJkOTk2MzE1NGRiYmNmZjkzM2Q3NjdjZTc3MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jz_jps8QJIU-_S--yi6oyAFQdPT94bFmBmgLoVXqetM">
> 
> Your code leaves the fans off even after ironing. I use this but it's not good anyway because the fans remain constant and don't stay off for example in the first layers or completely when printing PETG. A specific Orca function is required.
> 
> {if extrusion_role=~/Ironing/} M106 S0 ; disable fan M106 P2 S0 ; disable additional fan {else} M106 S[fan_max_speed] ; reactivate fans M106 P2 S[fan_max_speed] ; reactivate additional fan {endif}

Your  code causes fans to be overridden repeatedly, which can create issues, particularly when the filament is changing. For example, my Bambu Lab X1 Carbon has several states where the fans are set to different configurations, such as during bed leveling. Forcing the fans to run at maximum speed during every change can be problematic, especially for the fans that are constantly speeding up and slowing down. Additionally, this code interferes with the ability to adjust fan speeds manually from the control panel.

Here is my improved version:
```gcode
 {if extrusion_role=~/Ironing/}
M106 S0 ; turn off fan completely
M106 P2 S0 ; turn off additional fans completely
{endif}
{if extrusion_role=~/BottomSurface/}
M106 S0 ; turn off fan completely
M106 P2 S0 ; turn off additional fans completely
{endif}
{if last_extrusion_role=~/Ironing/ || last_extrusion_role=~/BottomSurface/}
M106 S255 ; reactivate fans to 100%
M106 P2 S255 ; reactivate additional fans to 100%
{endif}
```

Explanation:

This version turns off the fans during the "Ironing" or "BottomSurface" stages (you can adjust the fan levels as needed). It also checks whether the last mode was "Ironing" or "BottomSurface," and if so, reactivates the fans. Other modes work as they do by default, ensuring the system's flexibility and avoiding unnecessary fan overrides.

**Comment by jmmerida** [comment]
> > > In the meantime, this is how I did it automatically:
> > > <img alt="image" width="751" src="https://private-user-images.githubusercontent.com/20917687/375958031-ea019a5a-7f7b-4d2b-9f01-13818ba17f9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NzgzNDAsIm5iZiI6MTczMDY3ODA0MCwicGF0aCI6Ii8yMDkxNzY4Ny8zNzU5NTgwMzEtZWEwMTlhNWEtN2Y3Yi00ZDJiLTlmMDEtMTM4MThiYTE3ZjlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDIzNTQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjYTIxNDFmMWY3YjRlNGVhOTZiZWE2MjA5OWI4ODc5ZTMzNTJkOTk2MzE1NGRiYmNmZjkzM2Q3NjdjZTc3MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jz_jps8QJIU-_S--yi6oyAFQdPT94bFmBmgLoVXqetM">
> > 
> > 
> > Your code leaves the fans off even after ironing. I use this but it's not good anyway because the fans remain constant and don't stay off for example in the first layers or completely when printing PETG. A specific Orca function is required.
> > {if extrusion_role=~/Ironing/} M106 S0 ; disable fan M106 P2 S0 ; disable additional fan {else} M106 S[fan_max_speed] ; reactivate fans M106 P2 S[fan_max_speed] ; reactivate additional fan {endif}
> 
> Your code causes fans to be overridden repeatedly, which can create issues, particularly when the filament is changing. For example, my Bambu Lab X1 Carbon has several states where the fans are set to different configurations, such as during bed leveling. Forcing the fans to run at maximum speed during every change can be problematic, especially for the fans that are constantly speeding up and slowing down. Additionally, this code interferes with the ability to adjust fan speeds manually from the control panel.
> 
> Here is my improved version:
> 
>  {if extrusion_role=~/Ironing/}
> M106 S0 ; turn off fan completely
> M106 P2 S0 ; turn off additional fans completely
> {endif}
> {if extrusion_role=~/BottomSurface/}
> M106 S0 ; turn off fan completely
> M106 P2 S0 ; turn off additional fans completely
> {endif}
> {if last_extrusion_role=~/Ironing/ || last_extrusion_role=~/BottomSurface/}
> M106 S255 ; reactivate fans to 100%
> M106 P2 S255 ; reactivate additional fans to 100%
> {endif}
> Explanation:
> 
> This version turns off the fans during the "Ironing" or "BottomSurface" stages (you can adjust the fan levels as needed). It also checks whether the last mode was "Ironing" or "BottomSurface," and if so, reactivates the fans. Other modes work as they do by default, ensuring the system's flexibility and avoiding unnecessary fan overrides.

This messes with the first layer cooling. I was not able to disable the fans for the first layer.

**Comment by Fuzzyl0g1x** [comment]
> > > > In the meantime, this is how I did it automatically:
> > > > <img alt="image" width="751" src="https://private-user-images.githubusercontent.com/20917687/375958031-ea019a5a-7f7b-4d2b-9f01-13818ba17f9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA2NzgzNDAsIm5iZiI6MTczMDY3ODA0MCwicGF0aCI6Ii8yMDkxNzY4Ny8zNzU5NTgwMzEtZWEwMTlhNWEtN2Y3Yi00ZDJiLTlmMDEtMTM4MThiYTE3ZjlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAzVDIzNTQwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjYTIxNDFmMWY3YjRlNGVhOTZiZWE2MjA5OWI4ODc5ZTMzNTJkOTk2MzE1NGRiYmNmZjkzM2Q3NjdjZTc3MGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.jz_jps8QJIU-_S--yi6oyAFQdPT94bFmBmgLoVXqetM">
> > > 
> > > 
> > > Your code leaves the fans off even after ironing. I use this but it's not good anyway because the fans remain constant and don't stay off for example in the first layers or completely when printing PETG. A specific Orca function is required.
> > > {if extrusion_role=~/Ironing/} M106 S0 ; disable fan M106 P2 S0 ; disable additional fan {else} M106 S[fan_max_speed] ; reactivate fans M106 P2 S[fan_max_speed] ; reactivate additional fan {endif}
> > 
> > 
> > Your code causes fans to be overridden repeatedly, which can create issues, particularly when the filament is changing. For example, my Bambu Lab X1 Carbon has several states where the fans are set to different configurations, such as during bed leveling. Forcing the fans to run at maximum speed during every change can be problematic, especially for the fans that are constantly speeding up and slowing down. Additionally, this code interferes with the ability to adjust fan speeds manually from the control panel.
> > Here is my improved version:
> > {if extrusion_role=~/Ironing/}
> > M106 S0 ; turn off fan completely
> > M106 P2 S0 ; turn off additional fans completely
> > {endif}
> > {if extrusion_role=~/BottomSurface/}
> > M106 S0 ; turn off fan completely
> > M106 P2 S0 ; turn off additional fans completely
> > {endif}
> > {if last_extrusion_role=~/Ironing/ || last_extrusion_role=~/BottomSurface/}
> > M106 S255 ; reactivate fans to 100%
> > M106 P2 S255 ; reactivate additional fans to 100%
> > {endif}
> > Explanation:
> > This version turns off the fans during the "Ironing" or "BottomSurface" stages (you can adjust the fan levels as needed). It also checks whether the last mode was "Ironing" or "BottomSurface," and if so, reactivates the fans. Other modes work as they do by default, ensuring the system's flexibility and avoiding unnecessary fan overrides.
> 
> This messes with the first layer cooling. I was not able to disable the fans for the first layer.

Incorrect.  I have multiple printers and have not had a problem with this G: code.  It's 100% reliable, and should be standard in Orca.

**Comment by Noisyfox** [comment]
Please try PR [#9944](https://github.com/SoftFever/OrcaSlicer/pull/9944)
